### ISSUE TITLE
[explicit-function-return-type] allowExpressions and variable declarators

### ISSUE THREAD

**Issue by 43081j** [typescript-eslint/typescript-eslint#666]
More of a question than a bug report (for now).

At some point in time, [this line](https://github.com/typescript-eslint/typescript-eslint/blob/34e7d1ea4d4f7ee9e71c07d20fad711e4dd29d6c/packages/eslint-plugin/src/rules/explicit-function-return-type.ts#L243) of code was introduced.

What this effectively means is the rule's `allowExpressions` has no effect on something like the following:

```ts
const foo: SomeType = () => undefined;
```

This is of course an arrow function expression and should be allowed (being that the option is named "allow expressions").

Is there any particular reason you explicitly apply the rule when the parent is a variable declarator?

**Comment by bradzacher** [comment]
"At some point in time" = the `allowExpressions` option has always worked this way since it was added in https://github.com/bradzacher/eslint-plugin-typescript/pull/125 by ***you***!!

`const foo = () => undefined;` does technically include an arrow function expression, however it is not, for the purposes of the option, considered an expression because the intention of the statement is clearly meant as a function declaration.

In your example, if you use the `allowTypedFunctionExpressions`, then your example of `const foo: SomeType = () => undefined;` should not report, because you are assigning the expression to a variable with an explicit type.

**Comment by 43081j** [comment]
Hilarious! 😂 I even explain in that pr roughly why I did that. 

This happens when you contribute all over the place.. memory loss. 

Thanks for the explanation and pointing that out. 

I hadn't realised that's what the later introduced option was achieving either.
